src/tools/make-headers.r

```diff
@@ -81,11 +82,45 @@ emit-proto: proc [proto] [
             emit-rlib ["extern " proto "; // " the-file]
         ][
             emit-out ["extern " proto "; // " the-file]
+            either "REBTYPE" = proto-parser/proto.id [
+               emit-fsymb ["    SYM_FUNC(" proto-parser/proto.arg.1 "), // " the-file]
+            ][
+               emit-fsymb ["    SYM_FUNC(" proto-parser/proto.id "), // " the-file]
+            ]
         ]
         proto-count: proto-count + 1
     ]
 ]
 
+emit-directive: func [
+    directive
+    /local position
+][
+    process-conditional directive proto-parser/parse.position :emit-out output-buffer
+    process-conditional directive proto-parser/parse.position :emit-fsymb fsymbol-buffer
+]
+
+process-conditional: func [
+    directive
+    dir-position
+    emit [function!]
+    buffer
+    /local position
+][
+    emit [
+        directive
+        ;;; " // " the-file " #" line-of head dir-position dir-position
+    ]
+
+    ; Minimise conditionals for the reader - unnecessary for compilation.
+    if all [
+        find/match directive "#endif"
+        position: find/last tail buffer "#if"
+    ][
+        rewrite-if-directives position
+    ]
+]
+
 process: func [file] [
     if verbose [probe [file]]
     data: read the-file: file
```

Comment:
`proto-parser/emit-directive: :emit-directive` seems to be needed here (You might want to put it right before `proto-parser/process data`, but github doesn't allow me to comment on that line), or no directives will be emit for function symbols.